packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m

```diff
@@ -395,19 +406,48 @@ - (void)picker:(PHPickerViewController *)picker
                                    NSData *_Nullable imageData, NSString *_Nullable dataUTI,
                                    UIImageOrientation orientation, NSDictionary *_Nullable info) {
                                  // maxWidth and maxHeight are used only for GIF images.
-                                 [self saveImageWithOriginalImageData:imageData
-                                                                image:localImage
-                                                             maxWidth:maxWidth
-                                                            maxHeight:maxHeight
-                                                         imageQuality:desiredImageQuality];
+                                 savedPath = [FLTImagePickerPhotoAssetUtil
+                                     saveImageWithOriginalImageData:imageData
+                                                              image:localImage
+                                                           maxWidth:maxWidth
+                                                          maxHeight:maxHeight
+                                                       imageQuality:desiredImageQuality];
+                                 [pathList addObject:savedPath];
+
+                                 dispatch_semaphore_signal(resultSemaphore);
                                }];
               }
             });
+            dispatch_semaphore_wait(resultSemaphore, DISPATCH_TIME_FOREVER);
+
+            [self handlePath:pathList resultCount:results.count];
           }
         }];
   }
 }
 
+/**
+ * Calls the correct method
+ *
+ * If the @c pathList count and the @c resultCount are equal then checks
```

Comment:
We don't need to mention implementation details of this method. Probably a simple:
`Handle the result based on the count of pathList.`

(Do we need resultCount? resultCount is the same as pathList.count, right? Maybe I'm missing something)

Reply:
I used `resultCount` to compare with `pathlist.count` to make sure `pathList` has all images in it.